src/main/java/io/github/eatmyvenom/litematicin/utils/Printer.java

```diff
@@ -712,16 +769,25 @@ private static boolean canPlaceFace(FacingData facedata, BlockState stateSchemat
     private static boolean printerCheckCancel(BlockState stateSchematic, BlockState stateClient,
             PlayerEntity player) {
         Block blockSchematic = stateSchematic.getBlock();
+        if (blockSchematic instanceof SeaPickleBlock && stateSchematic.get(SeaPickleBlock.PICKLES) >1) {
+            Block blockClient = stateClient.getBlock();
 
+            if (blockClient instanceof SeaPickleBlock && stateClient.get(SeaPickleBlock.PICKLES) != stateSchematic.get(SeaPickleBlock.PICKLES)) {
+                return blockSchematic != blockClient;
+            }
+        }
         if (blockSchematic instanceof SlabBlock && stateSchematic.get(SlabBlock.TYPE) == SlabType.DOUBLE) {
             Block blockClient = stateClient.getBlock();
 
             if (blockClient instanceof SlabBlock && stateClient.get(SlabBlock.TYPE) != SlabType.DOUBLE) {
                 return blockSchematic != blockClient;
             }
         }
-
-        if (stateClient.isAir()) // This is a lot simpler than below. But slightly lacks functionality.
+        Block blockClient = stateClient.getBlock();
+        if (blockClient instanceof SnowBlock && stateClient.get(SnowBlock.LAYERS) <3) {
+                return false;
+        }
+        if (stateClient.isAir() || stateClient.getBlock().getTranslationKey().contains((String) "water") || stateClient.getBlock().getTranslationKey().contains((String) "lava")) // This is a lot simpler than below. But slightly lacks functionality.
```

Comment:
If i'm correct, your intension is to not place blocks in water and don't try to break them. This line only does the trick for clicking on blocks (e.g. repeaters).

Reply:
placing blocks to replace water/lava requires 1. bypass your breaking code check, 2. bypass isAir()check. so those lines are needed to ignore snow layer(less than 3), and fluid types.